### ISSUE TITLE
Very slow debug stepping

### ISSUE THREAD

**Issue by DanTup** [dart-lang/webdev#800]
Using the code at https://github.com/DanTup/dwds_slow_step_repro, if I place a breakpoint on the if statement in `lib/main.dart` as shown here:

![Screenshot 2019-10-30 at 4 38 13 pm](https://user-images.githubusercontent.com/1078012/67879068-3b1ae400-fb34-11e9-95ae-95098ad0e596.png)

Then hit it with the debugger (running in Chrome, using latest master Flutter + dwds) and press Step Over, it takes a long time (6-10 seconds on MacBook Pro) to step. The response to the resume is very quick, but the resume event and pause event are quite slow.

```
[4:38:15] ==> {"id":"472","method":"resume","params":{"isolateId":"1","step":"Over"}}
[4:38:15] <== {"jsonrpc":"2.0","result":{"type":"Success"},"id":"472"}
[4:38:20] <== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"Resume","timestamp":1572453500314,"isolate":{"type":"@Isolate","id":"1","number":"1","name":"http://localhost:54362/:main()"}}}}
[4:38:22] <== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"PauseInterrupted","timestamp":1572453500336,"isolate":{"type":"@Isolate","id":"1","number":"1","name":"http://localhost:54362/:main()"},"topFrame":{"type":"Frame","index":0,"code":{"type":"@Code","id":"820","name":"build","kind":"Dart"},"location":{"type":"SourceLocation","script":{"type":"@Script","id":"475","uri":"package:hello_world/main.dart"},"tokenPos":1353},"vars":[{"type":"BoundVariable","name":"context","value":{"type":"@Instance","id":"{\"injectedScriptId\":1,\"id\":3452}","kind":"PlainInstance","class":{"type":"@Class","id":"classes|package:flutter/src/widgets/framework.dart|StatelessElement","name":"StatelessElement"},"valueAsStringIsTruncated":false},"declarationTokenPos":-1,"scopeStartTokenPos":null,"scopeEndTokenPos":null},{"type":"BoundVariable","name":"m","value":{"type":"@Instance","id":"objects/bool-true","kind":"Bool","class":{"type":"@Class","id":"classes|dart:core|Bool","name":"Bool"},"valueAsString":"true","valueAsStringIsTruncated":false},"declarationTokenPos":-1,"scopeStartTokenPos":null,"scopeEndTokenPos":null},{"type":"BoundVariable","name":"this","value":{"type":"@Instance","id":"{\"injectedScriptId\":1,\"id\":3453}","kind":"PlainInstance","class":{"type":"@Class","id":"classes|package:hello_world/main.dart|MyHomePage","name":"MyHomePage"},"valueAsStringIsTruncated":false},"declarationTokenPos":-1,"scopeStartTokenPos":null,"scopeEndTokenPos":null}],"kind":"Regular"}}}}
```

I couldn't repro this in a new `flutter create` project, even though I can't see any obvious difference. I deleted my `build`/`.dart_tool`/`pubspec.lock`/`.packages` etc. and re-tested to ensure it was nothing in there, but I still see the same behaviour.

**Comment by grouma** [comment]
I did some investigation. This isn't a `package:webdev` or `package:dwds` issue. This is either a Flutter or Chrome issue.

Here's what I did:
1) I ran the app without debugging. I opened the Chrome DevTools and set the breakpoint using source maps. Stepping using Chrome's DevTools is extremely slow as well.
2) I ran the app with `package:flutter_web` (pre merge) using webdev and `--no-injected-client`. I again set the breakpoint using source maps. Stepping again was extremely slow.
3) I repeated the above steps on Chrome Dev (v79). Stepping performance was even worse!
4) I then ran the application in Firefox. Stepping performance is not perfect but near instant.

I'd open a tracking issue on the Flutter web repo to get to the bottom of this because Flutter is tickling some kind of Chrome issue.
